Apps/Common/src/CacheProviders/DistributedCacheProvider.cs

```diff
@@ -45,7 +45,7 @@ public DistributedCacheProvider(IDistributedCache cache, string? keyPrefix = nul
         }
 
         /// <inheritdoc/>
-        public T? GetOrSet<T>(string key, Func<T> valueGetter, TimeSpan? expiry = null)
```

Comment:
This is a false warning by Rider and should be disabled

Reply:
as discussed on teams could you change the Func<T> to Func<T?> and make the return T?.

Reply:
And the return value should be checked before being cached.

Reply:
Sure thing, I will do.